keyboards/keebio/quefrency/keymaps/marianas/customLogic.c

```diff
@@ -0,0 +1,381 @@
+#include QMK_KEYBOARD_H
+#include "customLogic.h"
+#include "keymap.h"
+#include "keyDefinitions.h"
+#include "relativity.h"
+#include "timer.h"
+
+static int16_t fnTimer = 0;
+
+static uint16_t random_value = 26339;
+
+uint32_t layer_state_set_user(uint32_t state)
+{
+  switch (biton32(state))
+  {
+    case QWERTY:
+      rgblight_mode(9);
+      break;
+    case SQLMACROS:
+      rgblight_mode(RGBLIGHT_MODE_RAINBOW_MOOD+2);
+      break;
+    case FN_LAYER:
+      rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL+5);
+      break;
+  }
+  return state;
+}
+
+#define colTimeout 250
+
+static int16_t colonTimer = 0;
+
+bool custom_keys(uint16_t keycode, keyrecord_t *record)
+{
+  if (record->event.pressed)
+  {
+    switch(keycode)
+    {
+      case SM_UP:
+        for (int i = 0; i < 19; i++)
+        {
+          register_code(KC_UP);
```

Comment:
You can use `tap_code` here, intead.   It does the same thing, with an optional delay.

Reply:
Might make this change in the future, gonna leave it as it is at the moment.